extensions/cornerstone-dynamic-volume/src/panels/WorkflowPanel.tsx

```diff
@@ -27,34 +19,45 @@ const styles = {
   },
 }
 
-function WorkflowPanel({ commandsManager }) {
-  const [stageId, setStageId] = useState(0);
+function WorkflowPanel({ servicesManager }) {
+  const { workflowStagesService } = servicesManager.services;
+  const [stages, setStages] = useState(workflowStagesService.stages);
+  const [activeStage, setActiveStage] = useState(workflowStagesService.activeStage);
 
-  const runCommand = useCallback(
-    (commandName, commandOptions = {}) => {
-      return commandsManager.runCommand(commandName, commandOptions);
-    },
-    [commandsManager]
-  );
+  const handleClick = useCallback((stage) => {
+    workflowStagesService.setActiveStage(stage.id);
+  }, [workflowStagesService]);
+
+  useEffect(() => {
+    const { unsubscribe } = workflowStagesService.subscribe(
+      workflowStagesService.EVENTS.STAGES_CHANGED,
+      () => setStages(workflowStagesService.stages)
+    );
+
+    return () => {
+      unsubscribe();
+    }
+  }, [servicesManager]);
 
-  function handleClick({ id, hpStageId }) {
-    setStageId(id);
-    console.log(id, hpStageId);
+  useEffect(() => {
+    const { unsubscribe } = workflowStagesService.subscribe(
+      workflowStagesService.EVENTS.ACTIVE_STAGE_CHANGED,
+      () => setActiveStage(workflowStagesService.activeStage)
+    );
 
-    runCommand('setHangingProtocol', {
-      protocolId: 'default4D',
-      stageId: hpStageId
-    });
-  }
+    return () => {
+      unsubscribe();
+    }
+  }, [servicesManager]);
```

Comment:
These two hooks can be merged as they a common dependency (`servicesManager`).

Reply:
`WorkflowPanel` will be deleted soon and that code is now in `StepProgressDropdownWithService`. You can see the changes I made [here](https://github.com/lscoder/Viewers/pull/1/files) which should also be merged with `OHIF:feat/preclinical-4d-base` in the next days. Thank you!